[rpc] default construct rpc agent options based on the backend type

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #30208 [rpc] add default arg for init_method
* **#30201 [rpc] default construct rpc agent options based on the backend type**

Provide a default constructor so that users don't have to construct
RPC agent options. Also rename this to RPCBackend Options as suggested.

Differential Revision: [D18628698](https://our.internmc.facebook.com/intern/diff/D18628698/)

rohan-varma committed:
[wip][rpc] default construct rpc agent options based on the backend type

Provide a default constructor so that users don't have to construct
RPC agent options. Also rename this to RPCBackend Options as suggested.

Differential Revision: [D18628698](https://our.internmc.facebook.com/intern/diff/D18628698/)

[ghstack-poisoned]

xush6528 commented on torch/distributed/rpc/api.py:
```diff
@@ -74,6 +74,12 @@ def _init_rpc_backend(
+        # default construct a set of RPC agent options.
+        rpc_agent_options = rpc.backend_registry.construct_rpc_agent_options(
+            backend
+        )
```
Make sense.

construct_rpc_agent_options handlers should provide defaults in Python world.

xush6528 reviewed (changes requested):
Noticed your PR summary and you want to do the renaming as well.

rohan-varma committed:
Update on "[wip][rpc] default construct rpc agent options based on the backend type"


Provide a default constructor so that users don't have to construct
RPC agent options. Also rename this to RPCBackend Options as suggested.

Differential Revision: [D18628698](https://our.internmc.facebook.com/intern/diff/D18628698/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[wip][rpc] default construct rpc agent options based on the backend type"


Provide a default constructor so that users don't have to construct
RPC agent options. Also rename this to RPCBackend Options as suggested.

Differential Revision: [D18628698](https://our.internmc.facebook.com/intern/diff/D18628698/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[wip][rpc] default construct rpc agent options based on the backend type"


Provide a default constructor so that users don't have to construct
RPC agent options. Also rename this to RPCBackend Options as suggested.

Differential Revision: [D18628698](https://our.internmc.facebook.com/intern/diff/D18628698/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[wip][rpc] default construct rpc agent options based on the backend type"


Provide a default constructor so that users don't have to construct
RPC agent options. Also rename this to RPCBackend Options as suggested.

Differential Revision: [D18628698](https://our.internmc.facebook.com/intern/diff/D18628698/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[wip][rpc] default construct rpc agent options based on the backend type"


Provide a default constructor so that users don't have to construct
RPC agent options. Also rename this to RPCBackend Options as suggested.

Differential Revision: [D18628698](https://our.internmc.facebook.com/intern/diff/D18628698/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[wip][rpc] default construct rpc agent options based on the backend type"


Provide a default constructor so that users don't have to construct
RPC agent options. Also rename this to RPCBackend Options as suggested.

Differential Revision: [D18628698](https://our.internmc.facebook.com/intern/diff/D18628698/)

[ghstack-poisoned]

xush6528 reviewed (approved):
Thanks, Rohan!

mrshenli commented on torch/distributed/rpc/api.py:
```diff
@@ -70,35 +76,20 @@ def _init_rpc_backend(
-            )
+    if not rpc_backend_options:
+        # default construct a set of RPC agent options.
+        rpc_backend_options = rpc.backend_registry.construct_rpc_backend_options(
```
nice!

rohan-varma committed:
Update on "[rpc] default construct rpc agent options based on the backend type"


Provide a default constructor so that users don't have to construct
RPC agent options. Also rename this to RPCBackend Options as suggested.

Differential Revision: [D18628698](https://our.internmc.facebook.com/intern/diff/D18628698/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] default construct rpc agent options based on the backend type"


Provide a default constructor so that users don't have to construct
RPC agent options. Also rename this to RPCBackend Options as suggested.

Differential Revision: [D18628698](https://our.internmc.facebook.com/intern/diff/D18628698/)

[ghstack-poisoned]
